#201 Added SD podcast stream option. Added responsive feed dropdown menu.

dedeibel opened the pull request:
Fixes  #201

Feed dropdown is now built up dynamically by abstract datastructure.
Ignore IntelliJ .idea dir and .iml
Added tests for round_time method and clarified its purpose by name

Old feeds are still available and untouched - I compared the result with the live version.

MaZderMind commented on app/controllers/frontend/feeds_controller.rb:
```diff
@@ -1,75 +1,132 @@
   class FeedsController < FrontendController
     before_action :set_conference, only: %i(podcast_folder)
-    EXPIRE_FEEDS = 15
+    FEEDS_EXPIRY_DURATION = 15.minutes
```
dieses ruby ey /o\ :D

MaZderMind commented on app/controllers/frontend/feeds_controller.rb:
```diff
@@ -1,75 +1,132 @@
+      xml = Rails.cache.fetch([:podcast, params[:quality], events_max_age.to_i], expires_in: FEEDS_EXPIRY_DURATION) do
+        feed = Feeds::PodcastGenerator.new(
+          view_context,
+          title: 'recent events feed',
```
The Feed-Title should reflect the Quality, so that Feeds with different Content have a distinguishable name.

MaZderMind commented on app/controllers/frontend/feeds_controller.rb:
```diff
@@ -1,75 +1,132 @@
+      xml = Rails.cache.fetch([:podcast_archive, params[:quality], events_min_age.to_i], expires_in: FEEDS_EXPIRY_DURATION) do
+        feed = Feeds::PodcastGenerator.new(
+          view_context,
+          title: 'archive feed',
```
same here (add quality to title)

MaZderMind commented on app/controllers/frontend/feeds_controller.rb:
```diff
@@ -1,75 +1,132 @@
+      xml = Rails.cache.fetch([:podcast_folder, params[:quality], @conference, @mime_type]) do
+        feed = Feeds::PodcastGenerator.new(
+          view_context,
+          title: "#{@conference.title} (#{@mime_type})",
```
same here (add quality to title). Also adding the Mime-Type to the Titles seems wrong to me, because in the UI-Dropdown I actally selected a human readable name – this should be the same information used in the feed here.

MaZderMind commented on app/helpers/frontend/feeds_navigation_bar_structure_helper.rb:
```diff
@@ -0,0 +1,63 @@
+    # rendered by _navbar_feeds.haml
+    def feed_structure
+      menu = [
+          { :left =>{ :content => 'News', :href => '/news.atom' } },
```
I would really appreciate if we could shrink down the available options. They are not understandable for Joe Average user. For reference: this is the menu built by this code:
![screenshot from 2017-05-20 12-23-53](https://cloud.githubusercontent.com/assets/142237/26275038/23880bf2-3d58-11e7-9534-59064600e3fc.png)

I, despite knowing the code, would not now which feed I actually want. last two xears? last 100? webm or mp4 feed? Furthermore, the Feed named "RSS" is actually an RDF feed - does anyone still use RDF over RSS?

I'd suggest leaving all existing feeds intact but shrinking the menu down to:
- Video Podcast Feed (last 100 HD Videos, MP4)
- Video Archive Feed (all HD Videos, MP4)
- ----
- Audio Podcast Feed (last 100 MP3 Audios)
- Audio Archive Feed (all MP3 Audios)
- ----
- Per-Format-Feeds (all Formats in HD/SD for those who really want to choose for themselfs)

dedeibel commented:
Thanks for the comments.

- I added the quality to the feeds title
- I changed the highlight of the event's subformats to underline instead of orange like suggested
- About reducing the feed entries. I had given no thought about this since I don't know the initial idea of including them. My goal was simply to provide SD versions. Maybe it's best to create a new issues for this if it is a problem.
- About the "SD quality" links. I tried some things. I think the SD button idea would make it too prominent given the fact that the HQ version is kind of the default.

**Adding a simple "or":**

<img src="https://user-images.githubusercontent.com/26708/27254064-c57fa3c8-5380-11e7-8b7a-4bd3e3b903e6.png" width="405px"/>

**Adding bullet points to the downloadable entries:**

<img src="https://user-images.githubusercontent.com/26708/27254072-df7d23c2-5380-11e7-992f-df69ab4faf3b.png" width="498px"/>

MaZderMind commented:
I like the bullet as a separator. Now I see them as an independent item.
